task_10/toma.lobanova/src/main/java/com/hillel/tasks/StringTasks.java

```diff
@@ -0,0 +1,79 @@
+package com.hillel.tasks;
+
+public class StringTasks {
+
+  public String replaceEmail(String email) {
+    if (email == null) {
+      return null;
+    }
+    if (email.contains("@")) {
+      email = email.replace("@", "[ at ]");
+    }
+    if (email.contains(".")) {
+      email = email.replaceAll("\\.", "[ dot ]");
+    }
+    return email;
+  }
+
+  public String shortenWords(String sentence) {
+    if (sentence == null) {
+      return null;
+    }
+    sentence = sentence.trim();
+    String[] words = sentence.split(" ");
+    for (int i = 0; i < words.length; i++) {
+      words[i] = checkChangeWord(words[i]);
+    }
+    return String.join(" ", words);
+  }
+
+  private String checkChangeWord(String word) {
+    int wordLength = word.length();
+    word = word.toLowerCase();
+    if (wordLength >= 4) {
+      word = "" + word.charAt(0) + (wordLength - 2) + word.charAt(wordLength - 1);
+    }
+    return word;
+  }
+
+  public int countPalindromes(String words) {
+    if (words == null || words.equals("")) {
+      return 0;
+    }
+    int countPallindrom = 0;
+    String[] wordsArray = words.split("\\,");
+    for (int i = 0; i < wordsArray.length; i++) {
+      if (checkWord(wordsArray[i])) {
+        countPallindrom++;
+      }
+    }
+    return countPallindrom;
+  }
+
+  private boolean checkWord(String word) {
+    word = word.toLowerCase();
+    word = word.trim();
+    StringBuilder builder = new StringBuilder();
+    builder.append(word);
+    if (word.equals(builder.reverse().toString())) {
+      return true;
+    }
+    return false;
+  }
+
+  public int binaryToDecimal(String number) {
+    if (number == null || number.equals("")) {
+      return 0;
+    }
+    int binary = Integer.parseInt(number);
```

Comment:
Even more. I wanted that you'll take each symbol from the String by using **charAt** method and then **Character.getNumericValue** to convert a symbol to integer as it is